pkg/sql/vtable/information_schema.go

```diff
@@ -0,0 +1,148 @@
+	ORDINAL_POSITION         INT NOT NULL,
+	COLUMN_DEFAULT           STRING,
+	IS_NULLABLE              STRING NOT NULL,
+	DATA_TYPE                STRING NOT NULL,
+	CHARACTER_MAXIMUM_LENGTH INT,
+	CHARACTER_OCTET_LENGTH   INT,
+	NUMERIC_PRECISION        INT,
+	NUMERIC_PRECISION_RADIX  INT,
+	NUMERIC_SCALE            INT,
+	DATETIME_PRECISION       INT,
+	CHARACTER_SET_CATALOG    STRING,
+	CHARACTER_SET_SCHEMA     STRING,
+	CHARACTER_SET_NAME       STRING,
+	GENERATION_EXPRESSION    STRING,          -- MySQL/CockroachDB extension.
+	IS_HIDDEN                STRING NOT NULL, -- CockroachDB extension for SHOW COLUMNS / dump.
+	CRDB_SQL_TYPE            STRING NOT NULL  -- CockroachDB extension for SHOW COLUMNS / dump.
+);`
+
+// InformationSchemaAdministrableRoleAuthorizations describes the schema of the
+// information_schema.administrable_role_authorizations table.
+const InformationSchemaAdministrableRoleAuthorizations = `
+CREATE TABLE information_schema.administrable_role_authorizations (
+	GRANTEE      STRING NOT NULL,
+	ROLE_NAME    STRING NOT NULL,
+	IS_GRANTABLE STRING NOT NULL
+);`
+
+// InformationSchemaApplicableRoles describes the schema of the
+// information_schema.applicable_roles table.
+const InformationSchemaApplicableRoles = `
+CREATE TABLE information_schema.applicable_roles (
+	GRANTEE      STRING NOT NULL,
+	ROLE_NAME    STRING NOT NULL,
+	IS_GRANTABLE STRING NOT NULL
+);`
+
+// InformationSchemaColumnPrivileges describes the schema of the
+// information_schema.column_privileges table.
+const InformationSchemaColumnPrivileges = `
+CREATE TABLE information_schema.column_privileges (
+	GRANTOR        STRING,
+	GRANTEE        STRING NOT NULL,
+	TABLE_CATALOG  STRING NOT NULL,
+	TABLE_SCHEMA   STRING NOT NULL,
+	TABLE_NAME     STRING NOT NULL,
+	COLUMN_NAME    STRING NOT NULL,
+	PRIVILEGE_TYPE STRING NOT NULL,
+	IS_GRANTABLE   STRING
+);`
+
+// InformationSchemaSchemata describes the schema of the
+// information_schema.schemata table.
+const InformationSchemaSchemata = `
+CREATE TABLE information_schema.schemata (
+	CATALOG_NAME               STRING NOT NULL,
+	SCHEMA_NAME                STRING NOT NULL,
+	DEFAULT_CHARACTER_SET_NAME STRING,
+	SQL_PATH                   STRING
+);`
+
+// InformationSchemaTables describes the schema of the
+// information_schema.tables table.
+const InformationSchemaTables = `
+CREATE TABLE information_schema.tables (
+	TABLE_CATALOG      STRING NOT NULL,
+	TABLE_SCHEMA       STRING NOT NULL,
+	TABLE_NAME         STRING NOT NULL,
+	TABLE_TYPE         STRING NOT NULL,
+	IS_INSERTABLE_INTO STRING NOT NULL,
+	VERSION            INT
+);`
+
+var informationSchemaTables = []string{
+	InformationSchemaColumns,
+	InformationSchemaAdministrableRoleAuthorizations,
+	InformationSchemaApplicableRoles,
+	InformationSchemaColumnPrivileges,
+	InformationSchemaSchemata,
+	InformationSchemaTables,
+}
+
+var informationSchemaMap = map[string]*tree.CreateTable{}
+
+func init() {
```

Comment:
If this is just for testing scenarios, we don't want to pay the cost of creating the ASTs in our production code path. Can't we put the map and resolution code in our testing package? Are there other potential users of this elsewhere?